packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml

```diff
@@ -475,13 +479,30 @@ processors:
   - urldecode:
       field: _tmp.cs_headers
       ignore_missing: true
-  - kv:
-      field: _tmp.cs_headers
-      description: Parse request headers string into individual header fields.
-      target_field: http.request.headers
-      field_split: '\n'
-      value_split: ':'
-      ignore_missing: true
+  - script:
+      lang: painless
+      description: This script parses and standardizes the http headers
+      source: |
+        String header_string = ctx._tmp.cs_headers;
+        if (header_string != null) {
+          header_string = header_string.trim();
+          Map header_map = new HashMap();
+
+          // https://gist.github.com/vjt/06b28fbd988788c2a7a71c63dd9163be
+          def header_list = header_string.splitOnToken((String)(char)0x0a);
+
+          for (item in header_list) {
+            def kv = item.splitOnToken(':', 2);
+            if (kv.length == 2) {
+              String key = kv[0].toLowerCase().replace('-','_');
+              header_map.put(key, kv[1]);
+            }
```

Comment:
in your comment below you say that empty headers are added as well, but from the code it looks like they are not? Or is a header like `Header:` resulting in `kv.length = 2`?

Reply:
Aha, what I wanted to say is that if a headers field is sent by cloudfront, the elastic event will get the http.request.headers map, even if the headers field from cloudfront does not contain headers or the headers can not be parsed. From what I undertand, headers are always key/value, thus if not two elements after parsing, they are not added.